packages/next-intl/src/middleware/middleware.tsx

```diff
@@ -58,8 +58,16 @@ export default function createMiddleware(config: MiddlewareConfig) {
     const hasUnknownHost = configWithDefaults.domains != null && !domain;
 
     function getResponseInit() {
-      // Nothing yet
-      return undefined;
+      if (hasOutdatedCookie) {
+        request.headers.set(HEADER_LOCALE_NAME, locale);
+      }
+      const responseInit = {
+        request: {
+          headers: request.headers
+        }
```

Comment:
This is the fix that allows forwarding headers, right? Can you include a regression test in [middleware.test.tsx](https://github.com/amannn/next-intl/blob/main/packages/next-intl/test/middleware/middleware.test.tsx) that fails for the previous implementation and now passes?

Reply:
Yes, sure. But need some time, if it's fine for you.